File: CodeFile/Roman_to_Integer_13/Roman_to_Integer_01.js
//[1st]: LeetCode 13 Roman to Integer

//Using function return integer to solve this problem
var romanToInt = function(s) {

    //Storing letters and corresponding values to the object
    const dic = {
        "I" : 1,
        "V" : 5,
        "X" : 10,
        "L" : 50,
        "C" : 100,
        "D" : 500,
        "M" : 1000
    }

    //Calculate the total value
    let total = 0;

    //Using the for loop to iterate the Roman String
    for(let i = 0; i < s.length; i ++){

        let curr = s[i];
        let next = s[i + 1];

        //If the Curr Value is less than Next, we could dedcue the current value by total, otherwise, we can add the total.
        if(dic[curr] < dic[next])
        {

            total -= dic[curr];
            console.log(total);
        }
        else
        {

            total += dic[curr];
            console.log(total);
        }

    }
    console.log(total);
    return total;

};


romanToInt("VI");

File: CodeFile/Roman_to_Integer_13/Roman_to_Integer_01.java
import java.util.HashMap;
import java.util.Map;

//[1st]: LeedCode 13 Roman to Integer
/*
        "I" : 1,
        "V" : 5,
        "X" : 10,
        "L" : 50,
        "C" : 100,
        "D" : 500,
        "M" : 1000
 */


 class Solution {

    /*
     * Function Name: romanToInt Function
     * Description: The function will transfer Roman Code to Integer. By adding parameter, it will return an integer.
     * Author: Tian Yu (Edward) Shiung
     * Student ID: stianyu99
     * Code Editor: Visual Studio Code
     */
    public int romanToInt(String s) {

        //Using HashMap to store letters and corrseponding values to solve this problem
        Map<Character, Integer> dic = new HashMap<>();
        dic.put('I',1);
        dic.put('V',5);
        dic.put('X',10);
        dic.put('L',50);
        dic.put('C',100);
        dic.put('D',500);
        dic.put('M',1000);

        //Calculate the total
        int total = 0;

        for(int i = 0; i < s.length(); i ++){

            char curr = s.charAt(i);
            char next = (i + 1 < s.length()) ? s.charAt(i + 1) : '\0';


            int currValue = dic.get(curr);
            int nextValue = (next != '\0') ? dic.get(next) : 0;

            if (currValue < nextValue) {
                total -= currValue;
            } else {
                total += currValue;
            }
        }

        System.out.println(total);
        return total;
    }


    public static void main(String[] args){
        new Solution().romanToInt("VI");
    }
};
